decorator.rb

```diff
@@ -0,0 +1,10 @@
+class NameDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
```

Comment:
- [ ] To make this work you will need to add an attribute accessor for `@nameable`.